openpgp/packet/reader.go

```diff
@@ -44,9 +44,17 @@ func (r *Reader) Next() (p Packet, err error) {
 			continue
 		}
 		// TODO: Add strict mode that rejects unknown packets, instead of ignoring them.
-		if _, ok := err.(errors.UnknownPacketTypeError); !ok {
-			return nil, err
+		if _, ok := err.(errors.UnknownPacketTypeError); ok {
+			continue
+		}
+		if _, ok := err.(errors.UnsupportedError); ok {
+			switch p.(type) {
+			case *SymmetricallyEncrypted, *AEADEncrypted, *Compressed, *LiteralData:
```

Comment:
Is it intended that this will re-throw not only on 
```errors.UnsupportedError("Symmetrically encrypted packets without MDC are not supported")```
but also on
```return errors.UnsupportedError("unknown SymmetricallyEncrypted version")```?

From the PR description it sounds so, just double checking to be sure

Reply:
Yeah, it's intended; the assumption is that there won't be two versions of the packet (one of which we support), for example, also because this is not legal in the message grammar of OpenPGP (unlike multiple signatures with different versions, for example, which is legal).